scipy/spatial/transform/_rotation.py

```diff
@@ -2389,34 +2473,43 @@ def align_vectors(
         array([[0., 1., 0.],
                [0., 1., 2.]])
         """
-        backend = backend_registry.get(array_namespace(a))
+        backend = backend_registry.get(array_namespace(a), xp_backend)
         q, rssd, sensitivity = backend.align_vectors(a, b, weights, return_sensitivity)
         if return_sensitivity:
-            return cls(q, normalize=False, copy=False), rssd, sensitivity
-        return cls(q, normalize=False, copy=False), rssd
+            return Rotation(q, normalize=False, copy=False), rssd, sensitivity
+        return Rotation(q, normalize=False, copy=False), rssd
 
+    @xp_capabilities(
+        jax_jit=False, skip_backends=[("array_api_strict", "pickling not supported")]
+    )
     def __getstate__(self) -> tuple[Array, bool]:
         return (self._quat, self._single)
 
+    @xp_capabilities(
+        jax_jit=False, skip_backends=[("array_api_strict", "pickling not supported")]
+    )
     def __setstate__(self, state: tuple[Array, bool]):
         quat, single = state
         xp = array_namespace(quat)
-        self._backend = backend_registry.get(xp)
+        self._backend = backend_registry.get(xp, xp_backend)
         self._quat = xp.asarray(quat, copy=True)
         self._single = single
 
     @property
+    @xp_capabilities()
     def single(self) -> bool:
         """Whether this instance represents a single rotation."""
         return self._single or self._quat.ndim == 1
 
+    @xp_capabilities()
     def __bool__(self) -> bool:
         """Comply with Python convention for objects to be True.
 
         Required because `Rotation.__len__()` is defined and not always truthy.
```

Comment:
out of scope: this logic makes no sense to me. if Rotation is a Container then it should return False when empty.

Reply:
Agreed. However, this would lead to an error for single rotations (`__len__` throws on `_single`). The proper way would be to return `self._quat.shape[0] != 0`. Leaving this unresolved for future reference